randovania/games/prime1/json_data/Chozo Ruins.txt

```diff
@@ -340,18 +342,35 @@ Extra - asset_id: 1014518864
 > Pit; Heals? False
   > Door to Ruined Shrine Access
       Any of the following:
-          Morph Ball or Space Jump Boots
+          Space Jump Boots
           All of the following:
-              Standable Terrain (Intermediate)
-              Combat/Scan Dash (Intermediate) or L-Jump (Intermediate)
+              Combat/Scan Dash (Intermediate) and Standable Terrain (Intermediate)
+              Scan Visor or Before Beetle Battle
+          L-Jump (Intermediate) and Standable Terrain (Intermediate)
+          All of the following:
+              Morph Ball
+              Any of the following:
+                  After Beetle Battle
```

Comment:
iirc, the invisible wall blocking the tunnel goes away after getting the pickup, not the fight.

Reply:
True, but then I'd have to add a 2nd event just for that. If you kill the plated beetle, you can escape through the hole in 100% of use cases

Reply:
Hmmmmm, unless it's gravity suit and you don't have bombs yet..... ughhhhhhhhhhhhhhh

Reply:
@dyceron see the latest commit, I think it solves the issue here

Reply:
No it doesn't. There is no connection from Pit -> Beetle Battle, so logically you'd need tricks or Space Jump to reach the item.

- Pit should connect to Morph Ball with either SJ, bombs or after Beetle Battle.
- Beetle Battle should reconnect to Pit and only Pit

Reply:
This last commit feels right. Sorry about the back and forth.

Reply:
It looks good, but to keep consistency with other event+pickup nodes, swap the event and pickup orders. Nodes that connect to Morph should connect to the Morph Event, and the Morph Event connects to Morph as Trivial, with Morph connecting to the other nodes. This is whats done in the other Primes as well.

Reply:
I did it this way because the generator shies away from events when it has other options available.